source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -55,9 +55,11 @@ RedisCluster::RedisCluster(
           factory_context.clusterManager(), factory_context.api().timeSource())),
       registration_handle_(refresh_manager_->registerCluster(
           cluster_name_, redirect_refresh_interval_, redirect_refresh_threshold_,
-          failure_refresh_threshold_, host_degraded_refresh_threshold_, [&]() {
+          failure_refresh_threshold_, host_degraded_refresh_threshold_,
+          [&]() {
             redis_discovery_session_.resolve_timer_->enableTimer(std::chrono::milliseconds(0));
-          })) {
+          })),
+      time_source_(dispatcher_.timeSource()) {
```

Comment:
Can we have time_source_ in BaseDynamicClusterImpl or one of the other cluster base classes vs. having it in every derived class? That might be a bit simpler and cleaner?